### ISSUE TITLE
[Inductor][Schedule][Fusion] Ops were not fused due to the incorrect score_fusion_memory.

### ISSUE THREAD

**Issue by etaf** [pytorch/pytorch#153346]
### 🐛 Describe the bug

```
import torch
# device = "cpu" : all ops fused to one kernel
# devcie = "cuda"/"xpu": all ops fused to two kernel.
device = "cuda"
loss = torch.compile(torch.nn.CrossEntropyLoss(), backend="inductor")
input = torch.randn(3, 5, device=device)
target = torch.empty(3, dtype=torch.long, device=device).random_(5)
for i in range(3):
    output = loss(input, target)

```
In the above reproducer, two Triton kernels are generated for `torch.nn.CrossEntropyLoss()` on CUDA/XPU, but in theory, the two fusions could be combined into a single kernel. On the CPU, they can be properly fused into one kernel.

### Analysis:
There are four ops that are lowered from the `torch.nn.CrossEntropyLoss()`：
```
op0: SchedulerNode(ComputedBuffer)
op0.writes = [MemoryDep('buf0', c0, {c0: 3})]
op0.unmet_dependencies = []
...
op1: SchedulerNode(ComputedBuffer)
op1.writes = [MemoryDep('buf1', c0, {c0: 3})]
op1.unmet_dependencies = [MemoryDep('buf0', c0, {c0: 3})]
...
op2: SchedulerNode(ComputedBuffer)
op2.writes = [MemoryDep('buf2', 0, {})]
op2.unmet_dependencies =
    [   MemoryDep('buf0', 0, {}),
        MemoryDep('buf0', 1, {}),
        MemoryDep('buf0', 2, {}),
        MemoryDep('buf1', 0, {}),
        MemoryDep('buf1', 1, {}),
        MemoryDep('buf1', 2, {})]
...
op3: SchedulerNode(ComputedBuffer)
op3.writes = [MemoryDep('buf3', 0, {})]
op3.unmet_dependencies = [MemoryDep('buf2', 0, {})]

```

The first round of  inductor scheduler fusion:  fuse op0 + op1 => op0_op1, op2 + op3=> op2_op3
```

op0_op1: FusedSchedulerNode(SchedulerNode,SchedulerNode)
op0_op1.writes = [MemoryDep('buf0', c0, {c0: 3}), MemoryDep('buf1', c0, {c0: 3})]
op0_op1.unmet_dependencies = []

op2_op3: FusedSchedulerNode(SchedulerNode,SchedulerNode)
op2_op3.writes = [MemoryDep('buf2', 0, {}), MemoryDep('buf3', 0, {})]
op2_op3.unmet_dependencies =
    [   MemoryDep('buf0', 0, {}),
        MemoryDep('buf0', 1, {}),
        MemoryDep('buf0', 2, {}),
        MemoryDep('buf1', 0, {}),
        MemoryDep('buf1', 1, {}),
        MemoryDep('buf1', 2, {})]

```
From the above IR, it's easy to conclude that op0_op1 and op2_op3 are fusible from the memory dependency perspective. However, the fusion score function score_fusion_memory() in the Inductor scheduler gives a score of 0.
The score_fusion_memory() function only checks whether two MemoryDep entries are exactly equal to determine if a read and a write match. In reality, it should be sufficient if the read is a subset of the write.

https://github.com/pytorch/pytorch/blob/c51bdf5acfb6a7abf3c8d908c451c92958e3e884/torch/_inductor/scheduler.py#L3873-L3878

Therefore, we need to improve the algorithm inside ·score_fusion_memory()· to allow fusion when the read is a subset of the write, rather than requiring an exact match.

### Versions

PyTorch version: 2.8.0a0+git1798b0d
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.31.2
Libc version: glibc-2.35

Python version: 3.10.15 | packaged by conda-forge | (main, Oct 16 2024, 01:24:24) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-5.15.47+prerelease24.6.13-x86_64-with-glibc2.35
